File: power-square-android/app/src/main/java/com/okla/ops/beans/NewCabinetBean.kt
package com.okla.ops.beans

/**
 * @Auther Administrator
 * @Date 2022/7/8 16:52
 * @Describe:
 */
data class NewCabinetBean(
    val stationModel: String,
    val stationName: String,
    val stationModelName: String,//电柜 规格仓数(用于storeNum字段)
    val standardImg: String,
    val address: String,
    val pid: String,
    val sn: String
)

File: lib/data/models/new_cabinet_bean.dart
import 'package:flutter/foundation.dart';

@immutable
class NewCabinetBean {
  final String? stationModel;
  final String? stationName;
  final String? stationModelName;
  final String? standardImg;
  final String? address;
  final String? pid;
  final String? sn;

  const NewCabinetBean({
    this.stationModel,
    this.stationName,
    this.stationModelName,
    this.standardImg,
    this.address,
    this.pid,
    this.sn,
  });

  factory NewCabinetBean.fromJson(Map<String, dynamic> json) {
    return NewCabinetBean(
      stationModel: json['stationModel']?.toString(),
      stationName: json['stationName']?.toString(),
      stationModelName: json['stationModelName']?.toString(),
      standardImg: json['standardImg']?.toString(),
      address: json['address']?.toString(),
      pid: json['pid']?.toString(),
      sn: json['sn']?.toString(),
    );
  }

  Map<String, dynamic> toJson() => {
        'stationModel': stationModel,
        'stationName': stationName,
        'stationModelName': stationModelName,
        'standardImg': standardImg,
        'address': address,
        'pid': pid,
        'sn': sn,
      };
}
